bugbug/model.py

```diff
@@ -70,26 +69,25 @@ def bugs():
 
         # Split dataset in training and test.
         X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
-
-        if self.undersampling_enabled:
-            # Under-sample the majority classes, as the datasets are imbalanced.
-            X_train, y_train = RandomUnderSampler(random_state=0).fit_sample(X_train, y_train)
-
-        print(f'X_train: {X_train.shape}, y_train: {y_train.shape}')
-        print(f'X_test: {X_test.shape}, y_test: {y_test.shape}')
+        pipeline = make_pipeline(self.sampler, self.clf)
 
         # Use k-fold cross validation to evaluate results.
         if self.cross_validation_enabled:
             scorings = ['accuracy', 'precision', 'recall']
-            scores = cross_validate(self.clf, X_train, y_train, scoring=scorings, cv=5)
+            scores = cross_validate(pipeline, X_train, y_train, scoring=scorings, cv=5)
+
             print('Cross Validation scores:')
             for scoring in scorings:
                 score = scores[f'test_{scoring}']
                 print(f'{scoring.capitalize()}: f{score.mean()} (+/- {score.std() * 2})')
 
-        # Evaluate results on the test set.
+        # Training on the resampled dataset
+        X_train, y_train = self.sampler.fit_resample(X_train, y_train)
```

Comment:
Is there a difference between `fit_sample` and `fit_resample`?

Reply:
It seems `fit_sample` is an alias for `fit_resample` for backward compatibility. It was called 'fit_sample` in earlier versions.